airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/steps/gradle.py

```diff
@@ -277,14 +202,16 @@ async def _collect_test_results(self, gradle_container: Container) -> Optional[A
         )
         if (
             test_results_dir_name_in_container
-            not in await gradle_container.directory(f"{self.context.connector.code_directory}/build").entries()
+            not in await self.dagger_client.host().directory(f"{self.context.connector.code_directory}/build").entries()
```

Comment:
again for understanding, why did this change to dagger_client? Shouldn't the test output be retrieved from the started gradle test process?

Reply:
Dagger can access the host file system. To still handle the same type of objects we use for artifact uploads (dagger.File, dagger.Directory) I preferred to retrieve it the _dagger way_ instead of implementing something custom.

Reply:
got it. So dagger best practices is to access all the OS pieces using the dagger client?

Reply:
It's just a way to seamlessly handle file and directory from host and containers in the same way, so you can move/mount them around with the same primitives.